eng/pipelines/templates/steps/compilation.check.yml

```diff
@@ -0,0 +1,31 @@
+parameters:
+  - name: Artifact
+    type: object
+    default: {}
+  - name: ArtifactName
+    type: string
+    default: "not-specified"
+  - name: RetryLimit
+    type: number
+    default: 30
+
+steps:
+  - download: current
+    artifact: ${{parameters.ArtifactName}}-signed
+  - pwsh: |
+      $pomf = Get-ChildItem "$(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{parameters.Artifact.groupId}}/${{parameters.Artifact.name}}/*.pom"
+      while ($retries++ -lt ${{parameters.RetryLimit}}) {
+        Write-Host "mvn --no-transfer-progress -f $pomf compile"
+        mvn --no-transfer-progress -f $pomf compile
+        if ($LASTEXITCODE) {
+          if ($retries -ge ${{parameters.RetryLimit}}) {
+            exit $LASTEXITCODE
+          }
+          Write-Host "Install failed, retrying in 1 minute..."
+          sleep 60
+        } else {
+          break
+        }
+      }
+    workingDirectory: $(Pipeline.Workspace)/${{parameters.ArtifactName}}-signed/${{parameters.Artifact.groupId}}/${{parameters.Artifact.name}}
+    displayName: "Verify Package Compilation"
```

Comment:
Any reason this is done through a PowerShell task rather than the Maven task built into DevOps?

Reply:
We need to do retries on mvn compile, because if one package depends on the unreleased version of another package, and the developer ships everything at once, pipeline will break and will require manual retries.
And as far as I know, there's no "retries" for devop tasks

Reply:
Ah! Yes, I see this now that this will loop 30 times before completely failing